cli/jobs/single-step/java/iris/src/main/java/com/microsoft/aml/auth/AMLAuthentication.java

```diff
@@ -0,0 +1,95 @@
+package com.microsoft.aml.auth;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.microsoft.aad.adal4j.AuthenticationContext;
+import com.microsoft.aad.adal4j.AuthenticationResult;
+import com.microsoft.aad.adal4j.ClientCredential;
+import com.microsoft.ml.experiment.MlFlowExperiment;
+
+/**
+ * This class is used to authenticate with AD and retrieve access token to be used with MlFLow
+ * @author Mufy, Abe
+ * @Date 7/1/2022
+ */
+public class AMLAuthentication {
+
+	private static AMLAuthentication amlAuth;
+
+	private static final Logger log = LoggerFactory.getLogger(AMLAuthentication.class);
+
+	private AMLAuthentication() {
+	}
+
+	private final String TARGET_RESOURCE = "https://management.core.windows.net/";
+
+	private final String _tenantId = System.getenv("TENANT_ID");
+	private final String AUTHORITY = String.format("https://login.microsoftonline.com/%s/oauth2/v2.0/token", _tenantId);
+	private final String CLIENT_ID = System.getenv("CLIENT_ID");;
+	private final String SECRET_UNUSED_LEFT_FOR_CLARITY = System.getenv("AZUREMLFLOW_SP_CLIENT_SECRET");
+
+	private final String _region = System.getenv("REGION");
+	private final String _subscriptionId = System.getenv("SUB_ID");
+	private final String _resourceGroup = System.getenv("RESOURCE_GROUP");
+	private final String _workspaceName = System.getenv("WORKSPACE_NAME");
+	private final String TRACKING_URI = String.format(
+			"https://%s.api.azureml.ms/mlflow/v1.0/subscriptions/%s/resourceGroups/%s/providers/Microsoft.MachineLearningServices/workspaces/%s",
+			_region, _subscriptionId, _resourceGroup, _workspaceName);
+
+	public static AMLAuthentication getInstance() {
```

Comment:
Using Azure Identity package for auth might be a good options since this is the Azure standard package for auth https://docs.microsoft.com/en-us/java/api/overview/azure/identity-readme?view=azure-java-stable

Reply:
Interesting. @mufajjul thoughts?

Reply:
Adding the suggestion as a comment.